lldb/source/Plugins/Language/Swift/SwiftUnsafeTypes.cpp

```diff
@@ -263,6 +277,71 @@ bool SwiftUnsafeRawBufferPointer::Update() {
   return true;
 }
 
+class SwiftUnsafePointer final : public SwiftUnsafeType {
+public:
+  SwiftUnsafePointer(ValueObject &valobj, UnsafePointerKind kind);
+  bool Update() override;
+};
+
+SwiftUnsafePointer::SwiftUnsafePointer(ValueObject &valobj,
+                                       UnsafePointerKind kind)
+    : SwiftUnsafeType(valobj, kind) {}
+
+bool SwiftUnsafePointer::Update() {
+  if (!m_valobj.GetNumChildren())
+    return false;
+
+  // Here is the layout of Swift's Unsafe[Mutable]Pointer.
+  //
+  // ▿ Unsafe[Raw]Pointer
+  //   ▿ _rawValue : Int
+  //       - pointerValue : Int
+  //
+
+  CompilerType type = m_valobj.GetCompilerType();
+  if (!type.IsValid()) {
+    LLDB_LOG(GetLogIfAllCategoriesSet(LIBLLDB_LOG_DATAFORMATTERS),
+             "{0}: Couldn't get the compiler type for the "
+             "'Swift.UnsafePointer' ValueObject.",
+             __FUNCTION__, type.GetTypeName());
+    return false;
+  }
+
+  auto *type_system =
+      llvm::dyn_cast_or_null<TypeSystemSwift>(type.GetTypeSystem());
+  if (!type_system) {
+    LLDB_LOG(GetLogIfAllCategoriesSet(LIBLLDB_LOG_DATAFORMATTERS),
+             "{0}: Couldn't get {1} type system.", __FUNCTION__,
+             type.GetTypeName());
+    return nullptr;
+  }
+
+  CompilerType argument_type =
+      type_system->GetGenericArgumentType(type.GetOpaqueQualType(), 0);
+
+  if (argument_type.IsValid())
+    m_elem_type = argument_type;
+
+  ValueObjectSP pointer_value_sp(m_valobj.GetChildAtIndex(0, true));
+  if (!pointer_value_sp) {
+    LLDB_LOG(GetLogIfAllCategoriesSet(LIBLLDB_LOG_DATAFORMATTERS),
+             "{0}: Couldn't unwrap the 'Swift.Int' ValueObject named "
+             "'pointerValue'.",
+             __FUNCTION__);
+    return false;
+  }
+
+  addr_t addr = pointer_value_sp->GetValueAsUnsigned(LLDB_INVALID_ADDRESS);
```

Comment:
We should really convert GetValueAsUnsigned to return an optional at one point...